### PR TITLE
User can create a flashcard 

### PR DESCRIPTION
**Pull request by StevenNagata**
react-flashcards-issue1

### CODE REVIEW

File: src/flashcard-form.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import React from 'react'
+
+export default class FlashcardForm extends React.Component {
+
+  render() {
+    return (
+      <div className="container-fluid w-50 border border-dark p-4 rounded">
```
Comment (thebearingedge) [inline src/flashcard-form.js]: Dark borders are _super_ risky. I don't really recommend using them unless you are diligent about applying them everywhere. And even then, it's probably not going to look that great.

File: src/flashcard-form.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        <form onSubmit={this.props.saveFlashcard}>
+          <h3 className="pb-3">Create a Flash Card</h3>
+          <div className="form-group">
+            <label htmlFor="formGroupExampleInput">Question</label>
```
Comment (thebearingedge) [inline src/flashcard-form.js]: Match your `htmlFor` to the actual `id` of the input you're labeling.

File: src/index.html [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+  <meta charset="UTF-8" />
+  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+  <meta http-equiv="X-UA-Compatible" content="ie=edge" />
+  <title>React FlashCards</title>
+  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.1.3/css/bootstrap.css" />
+  <link href="https://fonts.googleapis.com/css?family=Caveat|Kalam" rel="stylesheet"/>
```
Comment (thebearingedge) [inline src/index.html]: If you really want a more custom feel to your UI, you can choose a theme from here: https://bootswatch.com/
And link it from here: https://cdnjs.com/libraries/bootswatch

Avoid custom styles for this project. We want to concentrate on functionality and, unfortunately, you'll be judged on how it looks even though you haven't studied design.

File: src/app.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+export default class App extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      flashcards: []
+    }
+    this.saveFlashcard = this.saveFlashcard.bind(this)
+  }
+
+  saveFlashcard(event) {
```
Comment (thebearingedge) [inline src/app.js]: Try to move most of this method into your form component. `saveFlashCard` shouldn't be concerned with `event`s and element values. Instead, define a method that simply receives a new `flashCard` object and adds it to state. Push the work of building this object down into the form component.

**Review by thebearingedge — CHANGES_REQUESTED** [review]

**Comment by StevenNagata** [comment]
react-flashcards-issue1

File: src/app.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  constructor(props) {
+    super(props)
+    this.state = {
+      flashcards: []
+    }
+    this.saveFlashcard = this.saveFlashcard.bind(this)
+  }
+
+  saveFlashcard(newCard) {
+    const flashcards = this.state.flashcards.map((card) =>
```
Comment (thebearingedge) [inline src/app.js]: You don't need to copy every object in the array, just the array itself so you can add an object.

File: src/app.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+      Object.assign({}, card)
+    )
+    flashcards.push(newCard)
+    this.setState({ flashcards })
+    event.target.reset()
```
Comment (thebearingedge) [inline src/app.js]: Let the form component do the resetting.

File: src/styles.css [hunk 7]
```diff
@@ -0,0 +1 @@
+
```
Comment (thebearingedge) [inline src/styles.css]: You can add this file when it's needed.

**Review by thebearingedge — CHANGES_REQUESTED** [review]

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**